app/services/members/concerns/cleaned_up.rb

```diff
@@ -0,0 +1,55 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2021 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+module Members::Concerns::CleanedUp
+  extend ActiveSupport::Concern
+
+  included do
+    prepend AfterPerform
+
+    protected
+
+    def cleanup(member)
+      Members::CleanupService
+        .new(member.principal, member.project_id)
+        .call
+    end
+  end
+
+  module AfterPerform
+    protected
+
+    def after_perform(service_call)
```

Comment:
The callbacks don't have access to results currently. We would have to rework the services so that the callbacks get access to the result, same as to the params.

Reply:
Ah you're right regarding `after_call` but what should work is

```
around_call do
  result = yield
  ...
end
```

Reply:
Works like you said :+1: